www/include/reporting/dashboard/csvExport/csv_HostLogs.php

```diff
@@ -62,111 +60,175 @@
 
 $centreon = $_SESSION["centreon"];
 
-/*
- * Checking session
- */
+// Checking session
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->query(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = '" . $pearDB->escape($oreon->user->user_id) . "'"
+    );
     if (!$res->rowCount()) {
         get_error('bad session id');
     }
 } else {
     get_error('need session id!');
 }
 
-/*
- * getting host id
- */
-isset($_GET["host"]) ? $id = htmlentities($_GET["host"], ENT_QUOTES, "UTF-8") : $id = null;
-isset($_POST["host"]) ? $id = htmlentities($_POST["host"], ENT_QUOTES, "UTF-8") : $id;
+// getting host id
+$hostId = filter_var(
+    $_GET['host'] ?? $_POST['host'] ?? null,
+    FILTER_VALIDATE_INT
+);
 
-/*
- * Getting time interval to report
- */
+$hosts = $centreon->user->access->getHostAclConf(null, 'broker');
+
+//checking if the user has ACL rights for this resource
+if (!$centreon->user->admin
+    && $hostId !== null
+    && !array_key_exists($hostId, $hosts)
+) {
+    echo '<div align="center" style="color:red">' .
+        '<b>You are not allowed to access this host</b></div>';
+    exit();
+}
+
+// Getting time interval to report
 $dates = getPeriodToReport();
 $start_date = htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date = htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$host_name = getHostNameFromId($id);
+$host_name = getHostNameFromId($hostId);
```

Comment:
CamelCase mandatory

Reply:
Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)